server/src/main/java/com/cloud/deploy/DeploymentPlanningManagerImpl.java

```diff
@@ -613,6 +453,208 @@ public DeployDestination planDeployment(VirtualMachineProfile vmProfile, Deploym
         return dest;
     }
 
+    private DeployDestination deployInVmLastHost(VirtualMachineProfile vmProfile, DeploymentPlan plan, ExcludeList avoids,
+            DeploymentPlanner planner, VirtualMachine vm, DataCenter dc, ServiceOffering offering, int cpuRequested, long ramRequested,
+            boolean volumesRequireEncryption) throws InsufficientServerCapacityException {
+        HostVO host = _hostDao.findById(vm.getLastHostId());
+        _hostDao.loadHostTags(host);
+        _hostDao.loadDetails(host);
```

Comment:
@weizhouapache this needs to be fixed for https://github.com/apache/cloudstack/issues/9033, in main/4.20